data/calculated_parser/functions.py

```diff
@@ -301,6 +322,57 @@ def deepsoundchannel(depth, latitude, temperature, salinity) -> np.ndarray:
     return depth[min_indices]
 
 
+def deepsoundchannelbottom(depth, latitude, temperature, salinity) -> np.ndarray:
+    """
+    Find and return the deep sound channel bottom (the second depth where
+    the speed of sound id equal to the speed at the sonic layer depth).
+
+    Required Arguments:
+        * depth: Depth in meters
+        * latitude: Latitude in degrees North
+        * temperature: Temperatures in Celsius
+        * salinity: Salinity
+    """
+
+    depth, latitude, temperature, salinity = __validate_depth_lat_temp_sal(
+        depth, latitude, temperature, salinity)
+
+    # Use masked array to quickly enable/disable data (see below)
+    sound_speed = np.ma.array(sspeed(depth, latitude, temperature, salinity), fill_value=np.nan)
+
+    min_indices = __find_depth_index_of_min_value(sound_speed)
+    
+    sound_speed.mask = __get_soniclayerdepth_mask(sound_speed, min_indices)
+
+    # Find sonic layer depth indices
+    max_indices = __find_depth_index_of_max_value(sound_speed)
+
+    # Extract sound speed values for later comparison.
+    sound_speed_values_at_sonic_layer_depth = np.take_along_axis(
+        sound_speed,
+        max_indices[np.newaxis, :], # pad to equate number of dims to sound_speed
+        0 # apply along depth axis
+    )
+
+    # Flip the mask since we actually want to examine the values BELOW the sonic
+    # layer depth.
+    sound_speed.mask = ~sound_speed.mask
+
+    # Find Deep Sound Channel Bottom
+    dscb_values = np.isclose(sound_speed, sound_speed_values_at_sonic_layer_depth)
```

Comment:
https://docs.scipy.org/doc/numpy/reference/generated/numpy.isclose.html

`rtol` default is 1e-05.

Reply:
Thanks for taking the time to vectorize!! :)

I'm not sure about using `np.isclose()` because the sound speed profile is defined at discrete points and might not ever get super close to the speed we are looking for but it might cross through that point. For example,

Depth  (m) |  Speed of Sound  (m/s) |
-------------|--------------------------|
0  | 1000 |
10 | 1010 |
20 | 1000 |
30 | 1006 |
40 | 1009 | 
50 | 1012 |
60 | 1015 |

The SLD is 10 m and the speed of sound at that point is 1010 m/s but later in the profile the speed of sound never gets "really" close to 1010 m/s.

The sound profile crosses through 1010 m/s between 40 and 50 m. 



How do you feel about interpolating using the points around where the crossing occurs?

Reply:
it seemed too easy now that I think about it LOL I need to think about this.